core/src/main/scala/kafka/coordinator/GroupMetadataManager.scala

```diff
@@ -659,10 +660,10 @@ class GroupMetadataManager(val brokerId: Int,
       config.offsetsTopicNumPartitions
   }
 
-  private def getMessageFormatVersionAndTimestamp(partition: Int): (Byte, Long) = {
+  private def getMessageFormatVersionAndTimestampIfReplicaIsLocal(partition: Int): (Byte, Long) = {
```

Comment:
Instead of making the name really long and awkward, we could change the method to return `Option` as @hachikuji suggested and document that it returns `None` if the replica is not local.